ql/src/java/org/apache/hadoop/hive/ql/plan/SparkWork.java

```diff
@@ -62,17 +64,20 @@
 
   private Map<BaseWork, BaseWork> cloneToWork;
 
-  public SparkWork(String name) {
-    this.name = name + ":" + (++counter);
+  public SparkWork(String queryId) {
+    this.queryId = queryId;
+    this.dagName = queryId + ":" + counter.getAndIncrement();
     cloneToWork = new HashMap<BaseWork, BaseWork>();
   }
 
 
   @Explain(displayName = "DagName")
   public String getName() {
-    return name;
+    return this.dagName;
   }
 
+  public String getQueryId() { return this.queryId; }
```

Comment:
Please do not use a oneliner here

Reply:
OK! Have corrected it in the third commit. THX^^